apis/s3/src/main/java/org/jclouds/s3/config/S3HttpApiModule.java

```diff
@@ -184,7 +184,7 @@ protected void bindErrorHandlers() {
    }
 
    protected void bindRequestSigner() {
-      bind(RequestAuthorizeSignature.class).to(RequestAuthorizeSignatureV2.class).in(Scopes.SINGLETON);
+      bind(RequestAuthorizeSignature.class).to(RequestAuthorizeSignatureV4.class).in(Scopes.SINGLETON);
```

Comment:
I am willing to entertain changing the default to V4 but this should at least be a separate change.  We also probably need to document how to easily use the older signer.

Reply:
Hi @gaul 
We will be raising this change in separate PR as suggested.
and could you please elaborate on how to document for older signer.

Reply:
I think we should do something like this:

https://github.com/gaul/jclouds-1/pull/new/s3/signer

Note that this does not work and needs to move some code around.

@nacx Does this seem like the correct approach?

Reply:
Hi @nacx 
Could you please provide feedback on above approach,

@gaul 
 Will you be completing the remaining change on this PR/move code around.

Reply:
The approach looks good @gaul. Probably the code moving needs to change to something [like this](https://github.com/apache/jclouds/blob/master/apis/openstack-keystone/src/main/java/org/jclouds/openstack/keystone/auth/config/AuthenticationModule.java#L61-L77). The selected lines are the ones that configure, in OpenStack, the version of the Keystone API to inject based on a property. A similar pattern could be used here.

Reply:
@nacx Thanks for the hint!

@jiffysrc Please test this -- I only looked at the unit tests.

Reply:
@gaul 
Tested, This seems fine.